kore/src/Kore/Equation/Simplification.hs

```diff
@@ -82,16 +76,17 @@ simplifyEquation
     -> simplifier (MultiAnd (Equation RewritingVariableName))
 simplifyEquation equation@(Equation _ _ _ _ _ _ _) =
     do
-        simplifiedResults <-
-            simplifyPattern leftWithArgument
-        Monad.when
-            (any (not . isTop . predicate) simplifiedResults)
+        simplifiedCond <-
+            Simplifier.simplifyCondition
+                SideCondition.top
+                (fromPredicate argument')
+        lift $ Monad.unless
+            ((isTop . predicate) simplifiedCond)
             (throwE equation)
```

Comment:
```suggestion
```
We're already doing this exact check on line 87.

Reply:
Is it clear that the check is the same for `simplifiedResults` and `simplifedCond`?

Reply:
What do you mean?